feat(genapi): init docs (#3636)
scaleway/docs-content: ai-data/generative-apis/concepts.mdx

---
meta:
  title: Generative APIs - Concepts
  description: This page explains all the concepts related to Generative APIs
content:
  h1: Generative APIs - Concepts
  paragraph: This page explains all the concepts related to Generative APIs
tags:
dates:
  validation: 2024-08-27
categories:
  - ai-data
---

## API rate limits

API rate limits define the maximum number of requests a user can make to the Generative APIs within a specific time frame. Rate limiting helps to manage resource allocation, prevent abuse, and ensure fair access for all users. Understanding and adhering to these limits is essential for maintaining optimal application performance using these APIs.

## Context window

> The context window refers to the maximum amount of input data, often measured in tokens, that a generative model can process at once. This window size directly impacts the model's ability to maintain coherence and relevance in generating responses, especially in long-form content. For example, a larger context window allows the model to consider the preceding conversation or text more when generating its output.

Reviewer: Thanks, fixed

---
meta:
  title: Generative APIs - Concepts
  description: This page explains all the concepts related to Generative APIs
content:
  h1: Generative APIs - Concepts
  paragraph: This page explains all the concepts related to Generative APIs
tags:
dates:
  validation: 2024-08-27
categories:
  - ai-data
---

## API rate limits

API rate limits define the maximum number of requests a user can make to the Generative APIs within a specific time frame. Rate limiting helps to manage resource allocation, prevent abuse, and ensure fair access for all users. Understanding and adhering to these limits is essential for maintaining optimal application performance using these APIs.

## Context window

> The context window is the maximum amount of prompt data considered by the model to generate a response. Using models with high context length, you can provide more information to generate relevant responses. The context is measured in tokens.

Reviewer 2: ```suggestion
A context window is the maximum amount of prompt data considered by the model to generate a response. Using models with high context length, you can provide more information to generate relevant responses. The context is measured in tokens.
```

## Embeddings

Embeddings are numerical representations of text data that capture semantic information in a dense vector format. In Generative APIs, embeddings are essential for tasks such as similarity matching, clustering, and serving as inputs for downstream models. These vectors enable the model to understand and generate text based on the underlying meaning rather than just the surface-level words.

## Error handling

Error handling refers to the strategies and mechanisms in place to manage and respond to errors during API requests. This includes handling network issues, invalid inputs, or server-side errors. Proper error handling ensures that applications using Generative APIs can gracefully recover from failures and provide meaningful feedback to users.

## Parameters

Parameters are settings that control the behavior and performance of generative models. These include temperature, max tokens, and top-p sampling, among others. Adjusting parameters allows users to tweak the model's output, balancing factors like creativity, accuracy, and response length to suit specific use cases.

## Inter-token Latency (ITL)

> The inter-token latency corresponds to the average time elapsed between two generated tokens. It is usually expressed in milliseconds.

Reviewer 2: ```suggestion
The inter-token latency (ITL) corresponds to the average time elapsed between two generated tokens. It is usually expressed in milliseconds.
```

## Prompt Engineering

Prompt engineering involves crafting specific and well-structured inputs (prompts) to guide the model towards generating the desired output. Effective prompt design is crucial for generating relevant responses, particularly in complex or creative tasks. It often requires experimentation to find the right balance between specificity and flexibility.

## Retrieval Augmented Generation (RAG)

Retrieval Augmented Generation (RAG) is a technique that enhances generative models by integrating information retrieval methods. By fetching relevant data from external sources before generating a response, RAG ensures that the output is more accurate and contextually relevant, especially in scenarios requiring up-to-date or specific information.

## Stop words

> Stop words are a parameter set to tell the model to stop generating further tokens after one or more chosen tokens have been generated. This is useful for controlling the end of the model output, as it will cut off at the first occurrence of any of these strings.

Reviewer 3: ```suggestion
Stop words are a parameter set that tells the model to stop generating further tokens after one or more chosen tokens have been generated. This is useful for controlling the end of the model output, as it will cut off at the first occurrence of any of these strings.
```

Reviewer: Not sure. 
A parameter that tells
A parameter set (in order) to tell

Reviewer 3: oh i thought "set" was being used as a noun and not a verb. in this case 👍🏻
